apps/angular/src/core/services/authorization-api.service.ts

```diff
@@ -0,0 +1,102 @@
+import { ServerErrorMapper } from '@js-camp/core/mappers/server-error.mapper';
+import { LoginData } from '@js-camp/core/models/login-data';
+import { LoginDataMapper } from '@js-camp/core/mappers/login-data.mapper';
+import { AuthorizationTokens } from '@js-camp/core/models/authorization-tokens';
+import { AuthorizationTokensMapper } from '@js-camp/core/mappers/authorization-tokens.mapper';
+
+import { UrlConfigService } from './url-config.service';
+import { LocalStorageService } from './local-storage.service';
+
+/** Authorization API access service. */
+@Injectable({ providedIn: 'root' })
+export class AuthorizationApiService {
+
+	private readonly http = inject(HttpClient);
+
+	private readonly urlConfigService = inject(UrlConfigService);
+
+	private readonly localStorageService = inject(LocalStorageService);
+
+	/**
+	 * Registers a user.
+	 * @param registrationData - Date about the user required to create a profile.
+	 * @returns Authorization tokens.
+	 */
+	public register(registrationData: RegistrationData): Observable<AuthorizationTokens> {
+		return this.http.post<AuthorizationTokensDto>(this.urlConfigService.authorization.register,
+			RegistrationDataMapper.toDto(registrationData)).pipe(
+			map(response => AuthorizationTokensMapper.fromDto(response)),
+			tap(response => this.localStorageService.saveTokens(response)),
+			catchError((error: unknown) => {
+				if (error instanceof HttpErrorResponse) {
+					return this.handleError(error);
+				}
+				return throwError(() => error);
+			}),
+		);
+	}
+
+	/**
+	 * Authorizes a user.
+	 * @param loginData - Date about the user required to log in.
+	 * @returns Authorization tokens.
+	 */
+	public login(loginData: LoginData): Observable<AuthorizationTokens> {
+		return this.http.post<AuthorizationTokensDto>(this.urlConfigService.authorization.login,
+			LoginDataMapper.toDto(loginData)).pipe(
+			map(response => AuthorizationTokensMapper.fromDto(response)),
+			tap(response => this.localStorageService.saveTokens(response)),
+			catchError((error: unknown) => {
+				if (error instanceof HttpErrorResponse) {
+					return this.handleError(error);
+				}
+				return throwError(() => error);
+			}),
+		);
+	}
+
+	/**
+	 * Refresh access token with refresh token.
+	 * @returns Authorization tokens.
+	 */
+	public refreshAccessToken(): Observable<AuthorizationTokens> {
+		return this.localStorageService.getRefreshToken().pipe(
+			switchMap(refreshToken => {
+				if (refreshToken) {
+					return this.http.post<AuthorizationTokensDto>(this.urlConfigService.authorization.refresh,
+						{ refresh: refreshToken }).pipe(
+						map(response => AuthorizationTokensMapper.fromDto(response)),
+						tap(response => this.localStorageService.saveTokens(response)),
+						catchError((error: unknown) => throwError(() => error)),
+					);
+				}
+				return throwError(() => new Error('Failed to refresh token'));
+			}),
+			catchError((error: unknown) => throwError(() => error)),
```

Comment:
Let's improve this code a little
```suggestion
			switchMap(refreshToken => {
				if (refreshToken) {
					return this.http.post<AuthorizationTokensDto>(this.urlConfigService.authorization.refresh,
						{ refresh: refreshToken });
				}
				return throwError(() => new Error('Failed to refresh token'));
			}),
			map(response => AuthorizationTokensMapper.fromDto(response)),
			tap(response => this.localStorageService.saveTokens(response)),
```